guides/hack/56-modules/03-using-internal.md

```diff
@@ -0,0 +1,95 @@
+# Internal and module level visibility
+One goal of using modules is to separate a code unit's public and private interfaces. If a file is part of a module, its toplevel functions, classes, interfaces, traits, enums, and typedefs can be marked `internal`. An internal toplevel symbol cannot be referenced outside of its module. If a symbol is not internal, it is `public`, and it's part of the public API of the module. You can optionally use the `public` keyword on toplevel symbols to denote them as public. 
+
+```hack
+module foo;
+internal class Foo {}
+internal trait TFoo {}
+internal interface IFoo {}
+internal newtype TInternal = Foo;
+```
+
+```hack
+module bar;
+// error: Foo is internal, it cannot be referenced outside of its module
+public function foo(): Foo {
+}
+```
+
+Methods and properties within classes also gain a new visibility `internal`, which means that they can only be accessed or called within the module.
+```hack
+module foo;
+public class Foo {
+    internal function foo(): void {
+        echo "foo!\n";
+    }
+    internal int $prop = 5;
+    public function bar(): void {
+        echo "bar!\n";
+    }
+}
+```
+
+An internal method or property can be called anywhere from within a module. `internal` replaces the visibility keyword of the method or propertiy (i.e. `protected`, `public` or `private`). Note that **class members' visibilities do not have to match the visibility of the class itself**: an internal class can have public methods, and a public class can have internal methods. You can think of the visibility on a toplevel symbol to represent where a symbol is allowed to be denoted, whereas the visibility of a method or property to represent where that individual member can be accessed. 
```

Comment:
```suggestion
An internal method or property can be called anywhere from within a module. `internal` replaces the visibility keyword of the method or property (i.e. `protected`, `public` or `private`). Note that **class members' visibilities do not have to match the visibility of the class itself**: an internal class can have public methods, and a public class can have internal methods. You can think of the visibility on a toplevel symbol to represent where a symbol is allowed to be denoted, whereas the visibility of a method or property to represent where that individual member can be accessed. 
```

I don't understand "allowed to be denoted". Aren't both `FooInternal` and `$foo->internalProp` just access?